File: packages/contracts/ts/v1/order-request.ts
import type { OrderSide, OrderType } from './enums';

/**
 * Request to place a new order.
 */
export interface OrderRequest {
  order_id: string;
  user_id: string;
  contest_id: string;
  symbol: string;
  side: OrderSide;
  type: OrderType;
  qty: number;
  limit_price?: number;
  stop_price?: number;
  take_profit?: number;
  stop_loss?: number;
  client_ts: number;
}

File: packages/contracts/v1/order_request.go
package v1

// OrderRequest represents a request to place a new order.
type OrderRequest struct {
	OrderID    string    `json:"order_id"`
	UserID     string    `json:"user_id"`
	ContestID  string    `json:"contest_id"`
	Symbol     string    `json:"symbol"`
	Side       OrderSide `json:"side"`
	Type       OrderType `json:"type"`
	Qty        int64     `json:"qty"`
	LimitPrice *float64  `json:"limit_price,omitempty"`
	StopPrice  *float64  `json:"stop_price,omitempty"`
	TakeProfit *float64  `json:"take_profit,omitempty"`
	StopLoss   *float64  `json:"stop_loss,omitempty"`
	ClientTs   int64     `json:"client_ts"`
}
